backend/cfg/cfg.ml

```diff
@@ -504,7 +505,7 @@ let is_noop_move instr =
     (match instr.arg.(0).loc with
     | Unknown -> false
     | Reg _ | Stack _ -> Reg.same_loc instr.arg.(0) instr.res.(0))
-    && Proc.register_class instr.arg.(0) = Proc.register_class instr.res.(0)
+    && Proc.stack_slot_class_for instr.arg.(0) = Proc.stack_slot_class_for instr.res.(0)
```

Comment:
It needs separate cases here for `Reg` to check `register_class` and `Stack` to check `stack_slot_class`.